src/components/VideoPlayer/VideoPlayer.jsx

```diff
@@ -1,8 +1,32 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
 import { Grid, InnerGrid } from './styled';
 import { ButtonAtom, TextAtom } from '../../atoms';
+import PropTypes from 'prop-types';
+import { GlobalContext } from '../../states/GlobalContext';
 
 function VideoPlayer({ data }) {
+  const [state, dispatch] = useContext(GlobalContext);
+  const [saved, setSaved] = useState(false);
+
+  const index = state.favorites.findIndex(item => item.id === data.id);
+
+  function addFavorite() {
+    dispatch({ type: 'addFavorite', payload: data })
+    setSaved(true)
+  }
+
+  function removeFavorite() {
+    dispatch({ type: 'deleteFavorite', payload: index })
+    setSaved(false)
+  }
+
+  let button;
+  if (index >= 0 || saved) {
+    button = <ButtonAtom click={removeFavorite}>REMOVER DE FAVORITOS</ButtonAtom>
```

Comment:
You're mixing English and Spanish in the project. If you'd like to have both languages supported, consider adding i18n (internationalization) to it.